apis/logging/v1/input_receiver_types.go

```diff
@@ -6,7 +6,17 @@ package v1
 //
 // The fields of this struct define the set of known Receiver types.
 type ReceiverSpec struct {
-	HTTP *HTTPReceiver `json:"http,omitempty"`
+	// Port the Receiver listen on.
+	// +required
+	Port int32 `json:"port"`
```

Comment:
@alanconway Can we have a receiver without a listen port? A messaging subscriber that's given a URL, like https://vector.dev/docs/reference/configuration/sources/amqp ? Or an fd https://vector.dev/docs/reference/configuration/sources/file_descriptor?

Reply:
Yep, see above